org.lflang/src/org/lflang/generator/ReactionInstanceGraph.java

```diff
@@ -92,6 +92,14 @@ public void rebuild() {
             // Do not throw an exception so that cycle visualization can proceed.
             // throw new InvalidSourceException("Reactions form a cycle!");
         }
+        // FIXME: Is it OK to also assign deadlines here?
+        // Also do the traversal in the opposite order to calculate the deadlines
+        this.clear();
+        addNodesAndEdges(main);
+        // Assign a level to each reaction. 
+        // If there are cycles present in the graph, it will be detected here.
+        assignInferredDeadlines();
+
     }
```

Comment:
Can we check for no deadlines in the program.

Reply:
And not assign deadlines if so

Reply:
@edwardalee can I assume that the ReactionInstanceGraph always is connected to the top-level ReactorInstance? If that is true I can check whether there are any declaredDeadlines within the ReactorInstance. If it is not necessarily the top-level I would have to check somewhere else. If there is any deadline in the  program then we should assign inferred deadlines

Reply:
I think the answer to that is no? We make a ReactionInstanceGraph for each ReactorInstance. So we need a global flag saying this program has deadlines. Any idea on where to check it and where to store it?

Reply:
There is always a top-level ReactorInstance. 

There is code somewhere that searches globally to see whether there is a physical action. Perhaps that could be a model for code that checks for a deadline.